runtime/src/accounts.rs

```diff
@@ -955,6 +959,8 @@ pub fn create_test_accounts(
     }
 }
 
+// Only used by bench, not safe to call otherwise accounts can conflict with the
+// accounts cache!
 pub fn update_accounts(accounts: &Accounts, pubkeys: &[Pubkey], slot: u64) {
```

Comment:
nits: maybe rename to `update_accounts_for_bench` if it's not too tedious.

Reply:
makes sense!